### ISSUE TITLE
embedding_ops_test fails on AVX512 builds

### ISSUE THREAD

**Issue by markdryan** [tensorflow/tensorflow#21676]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**: N/A
- **TensorFlow installed from (source or binary)**: source
- **TensorFlow version (use command below)**: ('v1.8.0-6288-g335336a', '1.10.0-rc1')
- **Python version**: Python 2.7.12
- **Bazel version (if compiling from source)**: [bazel release 0.15.0]
- **GCC/Compiler version (if compiling from source)**: g++ (Ubuntu 5.4.0-6ubuntu1~16.04.10) 5.4.0 20160609
- **CUDA/cuDNN version**: N/A
- **GPU model and memory**: N/A 
- **Exact command to reproduce**:

bazel test --config=opt -- //tensorflow/python/kernel_tests:embedding_ops_test

### Describe the problem

The test case fails on AVX512 builds.  See the logs below.

### Source code / logs

```
======================================================================
FAIL: testTransform (__main__.EmbeddingLookupTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/user/.cache/bazel/_bazel_user/0c0d18d29bdc5ed463543206f26b0a12/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/python/kernel_tests/embedding_ops_test.runfiles/org_tensorflow/tensorflow/python/kernel_tests/embedding_ops_test.py", line 609, in testTransform
    self.assertAllEqual(simple, sharded)
  File "/home/user/.cache/bazel/_bazel_user/0c0d18d29bdc5ed463543206f26b0a12/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/python/kernel_tests/embedding_ops_test.runfiles/org_tensorflow/tensorflow/python/framework/test_util.py", line 1466, in assertAllEqual
    np.testing.assert_array_equal(a, b, err_msg=msg)
  File "/home/user/.local/lib/python2.7/site-packages/numpy/testing/nose_tools/utils.py", line 855, in assert_array_equal
    verbose=verbose, header='Arrays are not equal')
  File "/home/user/.local/lib/python2.7/site-packages/numpy/testing/nose_tools/utils.py", line 779, in assert_array_compare
    raise AssertionError(msg)
AssertionError:
Arrays are not equal

(mismatch 16.6666666667%)
 x: array([[2., 2., 2.],
       [2., 2., 2.],
       [2., 2., 2.],
       [2., 2., 2.]])
 y: array([[2., 2., 2.],
       [2., 2., 2.],
       [2., 2., 2.],
       [2., 2., 2.]])

----------------------------------------------------------------------
Ran 2 tests in 3.798s

FAILED (failures=1)
not equal where =  (array([2, 3]), array([0, 1]))
not equal lhs =  [2. 2.]
not equal rhs =  [2. 2.]
~
```

**Comment by markdryan** [comment]
I've submitted a [PR](https://github.com/tensorflow/tensorflow/pull/21677) that fixes the issue.  The underlying problem is that the existing code expects the results of two different floating point computations to match exactly.  In AVX512 builds one of the computations uses an optimised  Eigen algorithm for computing the square roots of vectors of doubles and this algorithm can produce slightly different results from the unoptimised version.  These differences cause the exact match to fail.  The PR fixes the issue by modifying the test case so that the results of the two computations only need to be close to each other, rather than matching exactly.

**Comment by michaelisard** [comment]
Thanks! I will leave this as contributions welcome until the PR is merged.

**Comment by markdryan** [comment]
Note //tensorflow/python/kernel_tests:embedding_ops_test_gpu is now also failing on master on AVX512 builds.  This unit test along with //tensorflow/python/kernel_tests:embedding_ops_test_gpu can be fixed by merging #21677.

**Comment by markdryan** [comment]
AFAIK, this is the last remaining AVX512 issue, now that #21265 has been merged.  It would be great to get this fixed for the next release.
